### ISSUE TITLE
[EPIC] Provide Zeebe REST API client and user task endpoints

### ISSUE THREAD

**Issue by koevskinikola** [camunda/zeebe#16166]
## Introduction

The second increment aims to complete the REST API exposure of the user task API on the Gateway/server side. It also aims to implement the REST API client in the existing Java client. With that, it unblocks the e2e implementation of user tasks topic, including IT/QA tests.

## Task Breakdown

The tasks here overlap with the Gateway and Client-related tasks of #15622. In the scope of this Epic, the `Complete task` REST endpoint will be implemented in the Gateway and Java Client, as well as a QA/IT test for the `Complete user task` API.

The remaining User task API endpoints will be implemented by the devs assigned to #15622 with supported from the devs assigned to this issue regarding OpenAPI/REST infrastructure details. 

```[tasklist]
### Gateway API
- [ ] https://github.com/camunda/zeebe/issues/16118
- [ ] https://github.com/camunda/zeebe/issues/16660
```

```[tasklist]
### Java client
- [ ] https://github.com/camunda/zeebe/issues/16184
- [ ] https://github.com/camunda/zeebe/issues/16588
- [ ] https://github.com/camunda/zeebe/issues/16538
- [ ] https://github.com/camunda/zeebe/issues/16558
```

### Out of scope

In the Zeebe Java client:
- Ignore authentication for now. It will be added before Zeebe `8.5` is released
  - https://github.com/camunda/zeebe/issues/16557
- Ignore metrics. We only need these metrics for job activation, which will be migrated to REST in `8.6`

## Other Teams

- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

## Links

- [Product Hub (internal)
](https://github.com/camunda/product-hub/issues/1926)
- [Cross-team collaboration doc (internal)](https://docs.google.com/document/d/1IxHvRBNa6VtQN53QjbHxfckD8mJqYxEuh5Uvza4pq80/edit)
- [Zeebe breakdown document (internal)](https://docs.google.com/document/d/1qN9gJHtgwli5n7GBqU-mm1GngS7-pQ38-i3v7_ivQVI/edit#heading=h.eqd411rw2pi9)

## Additional context

Blocked by https://github.com/camunda/zeebe/issues/15878

**Comment by aleksander-dytko** [comment]
@koevskinikola can we clarify the responsibility for Gateway and Client implementation? It seems duplicated with https://github.com/camunda/zeebe/issues/15622

**Comment by koevskinikola** [comment]
@aleksander-dytko how does it look like now?

**Comment by koevskinikola** [comment]
Closing as all the sub-tasks have been completed.
